Stx/v13.2.2

danisafta opened the pull request:
This is an important step for enabling CephFS on StarlingX. We need Metadata Servers to be up on each node that serves as monitor. In order to do that, we need to wait for the monitors to start before we can start any Metadata Server.

rchurch-wrs commented on src/init-ceph.in:
```diff
@@ -740,7 +740,27 @@ if [ "$command" = "stop" -o "$command" = "onestop" ]; then
+. /etc/platform/platform.conf
+
+if [ "$system_type" == "Standard" ];then
+        timeout 10 ceph -s
```
Use execute_ceph_cmd() to be consistent with other calls.

rchurch-wrs commented on src/init-ceph.in:
```diff
@@ -740,7 +740,27 @@ if [ "$command" = "stop" -o "$command" = "onestop" ]; then
+                if [ "$type" == "mds" ]; then
+                        continue
+                fi
+        what_out+=" $name"
```
Make sure indentation is correct

danisafta replied:
done

rchurch-wrs commented on src/init-ceph.in:
```diff
@@ -740,7 +740,27 @@ if [ "$command" = "stop" -o "$command" = "onestop" ]; then
     what="$new_order"
 fi
 
+# Check if the monitors are up before starting any mds
```
Expand on the comment on my we only need this on Standard installs vs AIOs

danisafta replied:
done

rchurch-wrs reviewed (commented):
Squash this change with the other so the pull request has only one commit

oponcea commented on src/init-ceph.in:
```diff
@@ -740,6 +740,25 @@ if [ "$command" = "stop" -o "$command" = "onestop" ]; then
+        if [ "$?" -ne 0 ]; then
+        for name in $what;
+        do
+                type=`echo $name | cut -c 1-3`
```
you can replace it with:
if [[ $name =~ ^mds\. ]]; then

format is e.g. mds.controller-0, isn't it?

danisafta replied:
yes, but I was trying tot keep style consistent with the other changes

oponcea commented on src/init-ceph.in:
```diff
@@ -740,6 +740,25 @@ if [ "$command" = "stop" -o "$command" = "onestop" ]; then
 fi
 
+# Check if the monitors are up before starting any mds
+what_out=
```
move this at the same level with what=what_out.

danisafta replied:
done

Commit:
Wait for ceph monitors to be up before starting MDS.

When working with cephFS we needs MDS. On Standard
deployments monitors need to be up before starting
MDS in order to avoid synchronization errors.
